lnwallet/channel_test.go

```diff
@@ -7601,6 +7616,29 @@ func TestIdealCommitFeeRate(t *testing.T) {
 		}
 	}
 
+	// maxFeeRate calculates the maximum feerate a channel is allowed to
+	// allocate to fees. This function prevents us from using contants and
+	// making it easier to reason about the test. We do not use the function
+	// `MaxFeeRate` explicitly.
+	maxFeeRate := func(c *LightningChannel,
+		maxFeeAlloc float64) chainfee.SatPerKWeight {
+
+		balance, weight := c.availableBalance(AdditionalHtlc)
+		feeRate := c.localCommitChain.tip().feePerKw
+		currentFee := feeRate.FeeForWeight(weight)
+
+		maxBalance := balance.ToSatoshis() + currentFee
+		expectedFee := math.Max(
+			float64(maxBalance)*maxFeeAlloc, float64(currentFee),
+		)
```

Comment:
Correct me if I'm wrong, but doesn't this just duplicate the logic implemented inside the LightningChannel object?

Reply:
yes you are right I changed it, now I use this function to calculate the max_feerate without expecting the ratcheting property